compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java

```diff
@@ -343,10 +343,11 @@ private void analyzeModuleInitFunc(BLangFunction funcNode) {
         this.currDependentSymbol.pop();
     }
 
-    private void checkForUninitializedGlobalVars(List<BLangSimpleVariable> globalVars) {
-        for (BLangSimpleVariable globalVar : globalVars) {
+    private void checkForUninitializedGlobalVars(List<BLangVariable> globalVars) {
+        for (BLangVariable globalVar : globalVars) {
             if (this.uninitializedVars.containsKey(globalVar.symbol)) {
-                this.dlog.error(globalVar.pos, DiagnosticCode.UNINITIALIZED_VARIABLE, globalVar.name);
+                this.dlog.error(globalVar.pos, DiagnosticCode.UNINITIALIZED_VARIABLE,
+                        ((BLangSimpleVariable) globalVar).name);
```

Comment:
There'll be scenarios where this cast could fail, right?

Reply:
The following compiles, even though it is invalid, and crashes at runtime.

```ballerina
[int, int] [a, b]; // no error

public function main() {
    int x = a; // no error
}
```

```log
[2020-11-11 00:05:47,501] SEVERE {b7a.log.crash} - null 
java.lang.NullPointerException
	at $_init.$gen$$0046$0060init$0062(.:1)
	at $_init.$moduleInit(.)
	at $_init.$lambda$$moduleInit$(.)
	at io.ballerina.runtime.internal.scheduling.SchedulerItem.execute(Scheduler.java:498)
	at io.ballerina.runtime.internal.scheduling.Scheduler.run(Scheduler.java:277)
	at io.ballerina.runtime.internal.scheduling.Scheduler.runSafely(Scheduler.java:245)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

Reply:
Yes, will look into that thanx

Reply:
fixed